### PR TITLE
Add TWBase64 interface exposing Base64 implementation

### PR DESCRIPTION
**Pull request by miloserdow**
Similarly to https://github.com/trustwallet/wallet-core/blob/master/include/TrustWalletCore/TWBase32.h, expose Base64 implementation from core library to Swift/Kotlin, issue: #2369

## How to test

* Run core tests
* Run Android tests
* Run iOS tests
* Run WASM tests

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add core tests
- [x] Add Android tests
- [x] Add Swift tests
- [x] Add WASM tests
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

### CODE REVIEW

File: tests/interface/TWBase64Tests.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+TEST(TWBase64, EncodeUrl) {
+    TW::Data data{'+', '\'', '?', 'a', 'b'};
+    auto encodedStr = TWBase64EncodeUrl(&data);
+    std::string result = TWStringUTF8Bytes(encodedStr);
+
+    ASSERT_EQ(result, "Kyc_YWI=");
+
+    TWStringDelete(encodedStr);
+}
```
Comment (Milerius) [inline tests/interface/TWBase64Tests.cpp:56]: New end of lines, If you use CLion, in editor options you can add a new lines at end of files on save.

File: include/TrustWalletCore/TWBase64.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+TWString *_Nonnull TWBase64Encode(TWData *_Nonnull data);
+
+/// Encode an input to Base64 with the alphabet safe for URL-s and filenames (RFC4648 with '-', '_')
+///
+/// \param data Data to be encoded (raw bytes)
+/// \return The encoded data
+TW_EXPORT_STATIC_METHOD
+TWString *_Nonnull TWBase64EncodeUrl(TWData *_Nonnull data);
+
+TW_EXTERN_C_END
```
Comment (Milerius) [inline include/TrustWalletCore/TWBase64.h:47]: Same remarks

**Review by catenocrypt — COMMENTED** [review]
Looks good so far.  Kotlin/Swift tests could be also added, isn't it?

**Comment by hewigovens** [comment]
@miloserdow what's the status of this one? do you need help to add mobile tests?

**Comment by miloserdow** [comment]
> @miloserdow what's the status of this one? do you need help to add mobile tests?

No, I'm fine, I'll wrap it up soon. Was focusing on something else

File: tests/interface/TWBase64Tests.cpp [hunk 3]
```diff
@@ -53,4 +53,5 @@ TEST(TWBase64, EncodeUrl) {
     ASSERT_EQ(result, "Kyc_YWI=");
 
     TWStringDelete(encodedStr);
-}
\ No newline at end of file
+}
+s
```
Comment (Milerius) [inline tests/interface/TWBase64Tests.cpp:57]: Spurious letter

**Review by Milerius — CHANGES_REQUESTED** [review]

**Comment by miloserdow** [comment]
Not sure what's wrong with this wasm test

**Comment by hewigovens** [comment]
Will take a look, it's passed locally? might be some difference between Linux / macOS

**Comment by miloserdow** [comment]
So there was the problem with wasm not being able to catch exceptions in C++ code properly which was causing the error. I changed our Base64 implementation and it should pass now.

**Review by catenocrypt — APPROVED** [review]
+1!

**Review by Milerius — APPROVED** [review]

**Review by hewigovens — APPROVED** [review]
nice work 🚀

**Pull request merged**